Fix #1300: Add/update licenseheaders.

calina-c opened the pull request:
Update licenseheaders for README files to 2024.
Add licenseheaders to new files.

@trentmc do you want to setup some configuration to do this automatically when new files are added/the year changes?

trentmc commented:
> @trentmc do you want to setup some configuration to do this automatically when new files are added/the year changes?

This would be amazing (assuming it's sufficiently lightweight)

calina-c commented:
> > @trentmc do you want to setup some configuration to do this automatically when new files are added/the year changes?
> 
> This would be amazing (assuming it's sufficiently lightweight)

I've investigated and it seems we can only do it via pre-commit hooks. Using the CI, there is no specific job and running the raw command does not show modifiable files, even if the `dry` option is present.

Regarding pre-commits, I've always been partial to pre-commits being part of our flow, but they are currently not, so I don't want to rattle any more cages regarding the process.

TL;DR: I updated the files with licenseheaders, added a template and instructions in dev.md on how devs can do this once a year. And that's it :)
